src/jit/lower.cpp

```diff
@@ -2742,6 +2742,7 @@ void Lowering::InsertPInvokeCallProlog(GenTreeCall* call)
 
         comp->fgMorphTree(helperCall);
         BlockRange().InsertBefore(insertBefore, LIR::SeqTree(comp, helperCall));
+        LowerNode(helperCall); // helper call is inserted before current node and should be lowered here.
```

Comment:
NVM, as you've pointed out, in that location we insert code later in the method than the call node we're currently lowering, so the new code will be lowered naturally.